core/src/main/resources/error/error-classes.json

```diff
@@ -1163,5 +1163,257 @@
     "message" : [
       "<msg>."
     ]
+  },
+  "_LEGACY_ERROR_TEMP_1000" : {
```

Comment:
just wondering why we choose use numbers than a descriptive way for the names?

E.g. `_LEGACY_ERROR_TEMP_1000` => `_LEGACY_ERROR_TEMP_LEGACY_STORE_ASSIGNMENT_POLICY_DISALLOWED`?

Reply:
Maybe it will cause too much work for the migration. If this does not bring much values please feel free to ignore.

Reply:
> The migration on temporary error classes allows to gather statistics about errors and detect most popular error classes.

@amaliujia Please, see in PR's description:

> Why are the changes needed?

The migration on temporary error classes allows to gather statistics about errors and detect most popular error classes.

Reply:
@amaliujia Also see https://github.com/apache/spark/pull/37916#issuecomment-1251888183

Reply:
Thanks for the context! I see it better now.